File: dotnet/tests/Buble.Sdk.Tests/FilesTests.cs
using Buble.Sdk.Files;
using Xunit;

namespace Buble.Sdk.Tests;

public sealed class FilesTests
{
    [Fact]
    public async Task UploadsMultipartFileAndFields()
    {
        var handler = new FakeHttpMessageHandler();
        handler.EnqueueJson("""{"data":{"id":"file_1","url":"https://example.test/file.png"}}""");
        using var client = new BubleClient(new BubleClientOptions
        {
            ApiKey = "sk_test",
            BaseUrl = "https://example.test",
            HttpClient = new HttpClient(handler)
        });

        await client.Files.UploadAsync(
            FileUpload.FromBytes(new byte[] { 1, 2, 3 }, "reference.png", "image/png"),
            new UploadOptions
            {
                FileType = "image",
                Model = "google/nano-banana",
                Mode = "image_to_image"
            });

        var body = handler.Requests[0].Body;
        Assert.Equal("/api/v1/files", handler.Requests[0].Uri.AbsolutePath);
        Assert.Contains("file_type", body);
        Assert.Contains("image_to_image", body);
        Assert.Contains("reference.png", body);
    }
}

File: swift/Tests/BubleTests/FilesTests.swift
import Buble
import XCTest

final class FilesTests: XCTestCase {
    func testUploadsMultipartFile() async throws {
        let transport = try TestHTTPTransport(responses: [
            .json([
                "data": [
                    "object": "file",
                    "provider": "r2",
                    "url": "https://example.com/source.png",
                    "key": "api/image/source.png",
                    "file_type": "image",
                    "content_type": "image/png",
                    "size": 3,
                    "filename": "source.png"
                ]
            ])
        ])
        let client = try makeClient(transport: transport)

        let uploaded = try await client.files.upload(
            .fromData(Data([1, 2, 3]), filename: "source.png", contentType: "image/png"),
            options: UploadOptions(fileType: "image", model: "google/nano-banana", mode: "image_to_image")
        )

        let requests = await transport.recordedRequests()
        let request = try XCTUnwrap(requests.first)
        let body = String(data: try XCTUnwrap(request.httpBody), encoding: .utf8) ?? ""
        XCTAssertEqual(uploaded.data.url.absoluteString, "https://example.com/source.png")
        XCTAssertTrue(body.contains("name=\"file_type\""))
        XCTAssertTrue(body.contains("image_to_image"))
        XCTAssertTrue(body.contains("filename=\"source.png\""))
    }
}
